### PR TITLE
testing/terraform: upgrade to 0.8.1

### PR DESCRIPTION
**Pull request by tboerger**
I have upgraded terraform to the latest stable release, I have applied the same variables and functions like @clandmeter got for hugo at https://github.com/tboerger/aports/blob/a7edb288d896e937ec5290fb51b8317c30e1c522/testing/hugo/APKBUILD to have it in a similar fashion.

### CODE REVIEW

File: testing/terraform/APKBUILD [hunk 1]
```diff
@@ -2,37 +2,36 @@
...
+pkgver=0.8.1
 pkgrel=0
 pkgdesc="Building, changing, and combining infrastructure safely and efficiently"
 url="https://www.terraform.io/"
-arch="x86_64"
+arch="all"
 license="MPL 2.0"
 depends=""
-makedepends="go"
+depends_dev=""
```
Comment (7heo) [inline testing/terraform/APKBUILD]: As @clandmeter said on IRC, having `$depends_dev` is useless if there's no `$subpackages`.

File: testing/terraform/APKBUILD [hunk 2]
```diff
@@ -2,37 +2,36 @@
...
+makedepends="$depends_dev go"
 install=""
+subpackages=""
```
Comment (7heo) [inline testing/terraform/APKBUILD]: As @clandmeter said on IRC, there's no need to declare `$subpackages` if empty.

File: testing/terraform/APKBUILD [hunk 3]
```diff
@@ -2,37 +2,36 @@
...
-	mkdir -p "$srcdir/src/github.com/hashicorp"
+	mkdir -p ${builddir%/*}
 	mv $srcdir/$pkgname-$pkgver "$builddir"/ || return 1
 	default_prepare
 }
 
 build() {
 	cd $builddir
-
-	GOPATH="$srcdir" go build -v -o bin/$pkgname || return 1
```
Comment (7heo) [inline testing/terraform/APKBUILD]: Why did you remove the `|| return 1`?

File: testing/terraform/APKBUILD [hunk 4]
```diff
@@ -2,37 +2,36 @@
...
+	export GOPATH="$srcdir"
+	go build -v -o bin/$pkgname
 }
 
 package() {
-	cd "$builddir"
-
-	install -Dm755 bin/$pkgname \
-		"${pkgdir}/usr/bin/$pkgname" || return 1
```
Comment (7heo) [inline testing/terraform/APKBUILD]: Why did you remove the `|| return 1`?

Comment (tboerger) [inline testing/terraform/APKBUILD]: Because of https://github.com/tboerger/aports/blob/6796bae33d8bcfcf234a8ded0fd58cced8a74d7b/testing/hugo/APKBUILD#L32

File: testing/terraform/APKBUILD (hunk 3 above)
Comment (tboerger) [inline testing/terraform/APKBUILD]: Added it again

File: testing/terraform/APKBUILD (hunk 4 above)
Comment (tboerger) [inline testing/terraform/APKBUILD]: Added it again

Comment (7heo) [inline testing/terraform/APKBUILD]: See previous comments thread.

**Comment by tboerger** [comment]
So, anything I still have to change?
